File: CodeForces/1343b/main.cpp
#include <iostream>

void solve() {
    int n;
    std::cin >> n;

    if((n / 2) % 2 == 1) {
        std::cout << "NO" << std::endl;
    } else {
        std::cout << "YES" << std::endl;
        
        int i = 0, c = 2, es = 0, os = 0;

        while(i < n/2)
        {
            std::cout << c << " ";
            c += 2;
            i += 1;
            es += c;
        }

        i = 0, c = 1;

        while(i < n/2 - 1)
        {
            std::cout << c << " ";
            c += 2;
            i += 1;
            os += c;
        }

        std::cout << es - os - 2 << std::endl;
    }

}

int main()
{
    int n;
    std::cin >> n;

    for(int i = 0; i < n; i++) {
        solve();
    }

    return 0;
}

File: CodeForces/1343b/main.py
def solve():
    n = int(input())
    if (n == 2):
        print('NO')
        return

    check = (n // 2) % 2

    if (check == 0):
        print("YES")
        arr = []

        i, c, es = 0, 2, 0
        while (i < n//2):
            print(c, end=' ')
            c += 2
            i += 1
            es += c

        i, c, os = 0, 1, 0
        while (i < n//2 - 1):
            print(c, end=' ')
            c += 2
            i += 1
            os += c

        print(es - os - 2)
        return

    else:
        print('NO')
        return


t = int(input())

for i in range(t):
    solve()
